dapps/digital-twin-detail/src/components/DigitalTwinDetail.vue

```diff
@@ -58,40 +58,107 @@
-                      <evan-button
-                        type="primary"
-                        :disabled="exporting"
-                        :isLoading="exporting"
-                        :label="$t('_twin-detail.data.context-menu.download-template')"
-                        @click="downloadTwinTemplate()"
-                      />
+                  </evan-swipe-panel>
+
+                  <evan-swipe-panel
+                    @init="duplicatePanel = $event"
+                    alignment="right"
+                    class="light"
+                    :title="'_twin-detail.data.context-menu.duplicate-twin' | translate"
+                    showBackdrop="true"
+                    type="default">
+                    <div
+                      class="d-flex h-100 align-items-center justify-content-center"
+                      v-if="exporting || duplicating">
+                      <evan-loading />
+                      <div class="mt-3 text-center">
+                        <h4 v-if="exporting">
+                          {{ `_twin-detail.data.context-menu.exporting-twin` | translate }}
+                        </h4>
+                        <h4 v-else-if="duplicating">
+                          {{ `_twin-detail.data.context-menu.duplicating-twin` | translate }}
+                        </h4>
+                      </div>
+                    </div>
+                    <evan-form-dbcp
+                      class="pt-5"
+                      ref="dbcpForm"
+                      :contractAddress="$store.state.twin.contractAddress"
+                      :description="$store.state.twin.description"
+                      i18nScope="_twin-detail.data.general"
+                      onlyForm="true"
+                      @init="dbcpForm = $event"
+                      v-else
+                    />
+                    <template slot="footer">
+                      <div class="d-flex">
+                        <evan-button
+                          type="secondary"
+                          :label="$t('_evan.cancel')"
+                          @click="duplicatePanel.hide()"
+                          class="mr-3"
+                        />
+                        <span class="mx-auto" />
+                        <evan-button
+                          type="primary"
+                          :disabled="exporting || duplicating"
+                          :label="$t('_twin-detail.data.context-menu.duplicate-twin')"
+                          @click="createTwinDuplicate()"
+                        />
+                      </div>
                     </template>
-                  </evan-modal>
+                  </evan-swipe-panel>
                 </div>
 
                 <evan-profile-picture
                   class="twin-avatar"
                   type="device"
-                  :src="'https://via.placeholder.com/96'"
+                  :src="$store.state.twin.description.imgSquare"
```

Comment:
is it sure that all keys always exists? Maybe you can wrap it into a getImageSrc function which handles missing fields smoothly in TS.

Reply:
It's really sure, that this path exists. It is loaded previously.